tests/instruments/test_instrument.py

```diff
@@ -150,6 +151,41 @@ class Fake(FakeInstrument):
     assert fake.x == 5
 
 
+def test_control_preprocess_reply():
+    # test setting preprocess_reply at property-level
+    class Fake(FakeInstrument):
+        x = Instrument.control(
+            "", "JUNK%d", "",
+            preprocess_reply=lambda v: v.replace('JUNK', ''),
+            cast=int
+        )
+
+    fake = Fake()
+    fake.x = 5
+    assert fake.read() == 'JUNK5'
```

Comment:
huh, I am confused why this works, i.e. why `read()` returns `JUNK5` - maybe add a comment explaining this?

Reply:
I will add a comment. `read` is used inside `ask`, which is used by `values`. Only in `values` the `preprocess_reply` is applied.

Reply:
ah, right.